src/shared/components/confirm_deletion.js

```diff
@@ -6,10 +6,7 @@ module.exports = React.createClass({
   },
 
   delete: function () {
-    const update_fn = this.props.payment
-      ? () => this.props.remove_payment(this.props.id)
-      : this.props.cancel_gift_aid
-    update_fn()
+    this.props.delete(this.props.id)
```

Comment:
@rjmk Only thing I am a little unsure about is that `this.props.id` is undefined for some of the instances of this component. Is that an issue?